### DISCUSSION TITLE
Route has an invalid POST export

### DISCUSSION THREAD

**Discussion by reeceatkinson** [vercel/next.js#62413] [App Router] [answered]
In my `route.ts` file I have this code to get the body from the POST request when I deploy to Vercel I get an error but in development it runs perfectly.

```
import { NextApiRequest, NextApiResponse } from 'next';
export async function POST(req: NextApiRequest, res: NextApiResponse) {
    if (req.method === "POST") {
         const { prompt } = req.body as { prompt: string };
```

Vercel Build Error:
```
Type error: Route "src/app/api/audio/route.ts" has an invalid "POST" export:
--
19:24:05.175 | Type "NextApiRequest" is not a valid type for the function's first argument.
19:24:05.175 | Expected "Request \| NextRequest", got "NextApiRequest".
19:24:05.175 |  
19:24:05.222 | Error: Command "npm run build" exited with 1
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

When you do `next build && next start` locally, you'll also see this error.

The problem is that Route Handlers in the App Router have a different signature.

They do not take a Response argument. Take a look here: https://nextjs.org/docs/app/building-your-application/routing/route-handlers, pay attention to the function signatures.

So your code should look a bit like this pseudocode:

```ts
import { type NextRequest, NextResponse} from 'next/server'
 
export async function POST(req: Request) {
  // the method is already POST!
  const data = await req.json();
  const { prompt } = data as { prompt: string }; // not a very good idea to cast like this...
```

**Reply by reeceatkinson** [reply]
Thank you so much @icyJoseph that worked!
